### PR TITLE
Fix ngen worker entrypoint handling of CPU count

### PR DESCRIPTION
**Pull request by robertbartel**
Fixing entrypoint script to properly account for hosts with more than a single CPU.  This includes the logic for supplying args to `mpirun`,  dealing with (or ignoring) partition config dataset details, and branching execution for parallel versus serial jobs.

### CODE REVIEW

**Comment by robertbartel** [comment]
Found something else related that needs to be tweaked in the image, so making this a draft for the moment.

File: docker/main/ngen/entrypoint.sh [hunk 1]
```diff
@@ -158,9 +168,17 @@ exec_serial_ngen_run()
...
 check_for_dataset_dir "${BMI_CONFIG_DATASET_DIR}"
-# Don't require a partitioning dataset when only using a single node
+# Require a partitioning dataset only when doing parallel, multiprocess ngen job with MPI
+# Most obvious indicator of parallel processing (and simplest to check) is if there are multiple host nodes
 if [ ${MPI_NODE_COUNT:?} -gt 1 ]; then
     check_for_dataset_dir "${PARTITION_DATASET_DIR:?No partition dataset directory defined}"
+# If using only a single node, then we must check how many CPUs the host has
+elif [ $(echo "${MPI_HOST_STRING}" | sed 's/,//' | awk -F: '{print $2}') -gt 1 ] 2>/dev/null ; then
+    check_for_dataset_dir "${PARTITION_DATASET_DIR:?No partition dataset directory defined}"
+# Also sanity the check host string format and CPU count extraction, ensuring we didn't produce a false negative above
```
Comment (PhilMiller) [inline docker/main/ngen/entrypoint.sh:178]:
```suggestion
# Also sanity check the host string format and CPU count extraction, ensuring we didn't produce a false negative above
```

**Comment by robertbartel** [comment]
Found something else that also needs related modification.

**Review by aaraney — APPROVED** [review]
Looks great! Thanks, @robertbartel

**Pull request merged**